aws/read_from_queue.py

```diff
@@ -1,25 +1,33 @@
 import json
+import uuid
 from os import getenv
 
 import boto3
+import redis
 from src.transformers.employee_transformer import transform
 
+redis_client = redis.StrictRedis(host=getenv('REDIS_HOST'), port=getenv('REDIS_PORT'))
 s3 = boto3.resource('s3', endpoint_url=getenv('AWS_ENDPOINT_URL'))
 bucket = s3.Bucket('feras-bucket')
 
 sqs_resource = boto3.resource('sqs', endpoint_url=getenv('AWS_ENDPOINT_URL'))
 queue = sqs_resource.get_queue_by_name(QueueName="sqs-queue")
-
-file = open("data/output.json", "a")
-
-while (int(queue.attributes['ApproximateNumberOfMessages'])
-       or int(queue.attributes['ApproximateNumberOfMessagesNotVisible'])):
+ids = []
+while (int(queue.attributes['ApproximateNumberOfMessages']) > 0
+       or int(queue.attributes['ApproximateNumberOfMessagesNotVisible'])) > 0:
     response = queue.receive_messages()
     line = response[0].body
     new_line = transform(json.loads(line))
-    file.write(f"{json.dumps(new_line)}\n")
+    id = uuid.uuid4().int
+    redis_client.set(id, json.dumps(new_line))
+    ids.append(id)
     response[0].delete()
 
 
+file = open("data/output.json", "a")
+for x in ids:
```

Comment:
do not maintain ids in memory, use a redis command that gets all keys and loop through them, write them to the file, and then remove them one by one instead of using `flushdb`. This doesn't mean that what you're doing is bad, we just want to use more of the redis API.

Reply:
changed